### PR TITLE
schdeuler: Set resource with step for collecting plan

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5539
If we determine  which stores cannot be targets at the beginning of the scheduling,  we should set resource with step. 

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/schedule/plan/plan.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type Plan interface {
 
 	Clone(ops ...Option) Plan // generate plan for clone option
 	SetResource(interface{})
+	SetResourceWithStep(interface{}, int)
```
Comment (bufferflies) [inline server/schedule/plan/plan.go:26]: how about use enum?

File: server/schedule/plan/plan.go [hunk 2]
```diff
@@ -83,3 +84,10 @@ func SetResource(resource interface{}) Option {
 		plan.SetResource(resource)
 	}
 }
+
+// SetResource is used to generate Resource for plan
+func SetResourceWithStep(resource interface{}, step int) Option {
```
Comment (rleungx) [inline server/schedule/plan/plan.go:89]: Where is it used?

Comment (CabinfeverB) [reply server/schedule/plan/plan.go:89]: https://github.com/tikv/pd/pull/5544 will need it

**Review by bufferflies — APPROVED** [review]
lgtm

**Review by rleungx — APPROVED** [review]

**Pull request merged**